src/test/java/com/github/tomakehurst/wiremock/client/ResponseDefinitionBuilderTest.java

```diff
@@ -156,4 +170,54 @@ void proxyResponseDefinitionWithExtraInformationIsInResponseDefinitionWithBinary
         equalTo(new HttpHeaders(List.of(new HttpHeader("header", "value")))));
     assertThat(proxyDefinition.getProxyUrlPrefixToRemove(), equalTo("/remove"));
   }
+
+  @Test
+  void responseDefinitionBuilderWithGzipDisabled() {
+    ResponseDefinition responseDefinition =
+        ResponseDefinitionBuilder.responseDefinition().withGzipDisabled(true).build();
+
+    assertNotNull(responseDefinition);
+    assertEquals(
+        "none", responseDefinition.getHeaders().getHeader(CONTENT_ENCODING).getValues().get(0));
+  }
+
+  @Test
+  void wireMockServerWithStubForWithGzipDisabledTrue() {
+    WireMockServer wireMockServer = new WireMockServer(options().dynamicPort().dynamicHttpsPort());
+    wireMockServer.start();
+    WireMockTestClient client = new WireMockTestClient(wireMockServer.port());
+
+    wireMockServer.stubFor(
+        get(urlEqualTo("/todo/items"))
+            .willReturn(
+                aResponse()
+                    .withStatus(200)
+                    .withResponseBody(
+                        new Body(
+                            "Here is some kind of response body"
+                                + "Here is some kind of response body"
+                                + "Here is some kind of response body"))));
+
+    WireMockResponse compressedResponse =
+        client.get("/todo/items", new TestHttpHeader("Accept-Encoding", "gzip"));
+
+    wireMockServer.stubFor(
+        get(urlEqualTo("/todo/items"))
+            .willReturn(
+                aResponse()
+                    .withStatus(200)
+                    .withGzipDisabled(true)
+                    .withResponseBody(
+                        new Body(
+                            "Here is some kind of response body"
+                                + "Here is some kind of response body"
+                                + "Here is some kind of response body"))));
+
+    WireMockResponse ordinaryResponse =
+        client.get("/todo/items", new TestHttpHeader("Accept-Encoding", "gzip"));
+
+    assertTrue(compressedResponse.content().length() < ordinaryResponse.content().length());
+    assertFalse(compressedResponse.content().contains("Here is some kind of response body"));
```

Comment:
This assertion might be better expressed as something like

```java
assertTrue(Gzip.isGzipped(compressedResponse.binaryContent()));
```

Reply:
done